apps/remix-ide/src/app/tabs/test-tab.js

```diff
@@ -52,18 +54,26 @@ module.exports = class TestTab extends ViewPlugin {
   }
 
   listenToEvents () {
-    this.on('filePanel', 'newTestFileCreated', file => {
-      var testList = this._view.el.querySelector("[class^='testList']")
-      var test = this.createSingleTest(file)
-      testList.appendChild(test)
+    this.on('filePanel', 'newTestFileCreated', async file => {
+      try {
+        await this.testTabLogic.getTests((error, tests) => {
+          if (error) return tooltip(error)
+          this.data.allTests = tests
+          this.data.selectedTests = [...this.data.allTests]
+          this.updateTestFileList(tests)
+          if (!this.testsOutput) return // eslint-disable-line
+        })
+      } catch (e) {
+        console.log(e)
+      }
       this.data.allTests.push(file)
```

Comment:
do you need to have `this.data.allTests.push(file)` and `this.data.selectedTests.push(file)` they seem to be updated in the callback above.